src/lib/draw/SplitRouteMode.svelte

```diff
@@ -50,13 +51,16 @@
       $map.unproject(e.point).distanceTo($map.unproject(nearbyPoint)) / 1000.0;
 
     // Are we snapped to anything?
-    let candidates = [];
+    let candidates: [number, Position, number][] = [];
     for (let [index, f] of $gjScheme.features.entries()) {
       if (f.geometry.type == "LineString") {
         let snapped = nearestPointOnLine(f.geometry, cursor, {
           units: "kilometers",
         });
-        if (snapped.properties.dist <= thresholdKm) {
+        if (
+          snapped.properties.dist != undefined &&
```

Comment:
The property is optional in the declared types, but scanning through the implementation, not sure when it's possible to be undefined. Couldn't figure out how to express this type narrowing more succinctly; I really dislike that the truthiness check in JS/TS treats 0 and "" as false